firefox-ios/Client/Frontend/Browser/WebView/WebviewViewController.swift

```diff
@@ -38,6 +61,42 @@ class WebviewViewController: UIViewController, ContentContainable, Screenshotabl
     func update(webView: WKWebView) {
         self.webView = webView
         setupWebView()
+        guard let documentLoadingView else { return }
+        view.bringSubviewToFront(documentLoadingView)
+    }
+
+    func showDocumentLoadingView() {
+        guard documentLoadingView == nil else { return }
+        let documentLoadingView = TemporaryDocumentLoadingView(frame: view.bounds)
+        documentLoadingView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(documentLoadingView)
+        NSLayoutConstraint.activate([
+            documentLoadingView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            documentLoadingView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            documentLoadingView.topAnchor.constraint(equalTo: view.topAnchor),
+            documentLoadingView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
```

Comment:
I recently learned we have this extension to `.pinToSuperview()`, could be great to use here too
```suggestion
        documentLoadingView.pinToSuperview()
```

Reply:
wow so helpful this thanks @lmarceau 😃